app/src/main/java/com/github/polybooks/core/database/implementation/OLBookDatabase.kt

```diff
@@ -0,0 +1,259 @@
+package com.github.polybooks.core.database.implementation
+
+import android.annotation.SuppressLint
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.github.polybooks.core.Book
+import com.github.polybooks.core.Interest
+import com.github.polybooks.core.database.DatabaseException
+import com.github.polybooks.core.database.interfaces.BookDatabase
+import com.github.polybooks.core.database.interfaces.BookOrdering
+import com.github.polybooks.core.database.interfaces.BookOrdering.*
+import com.github.polybooks.core.database.interfaces.BookQuery
+import com.google.firebase.Timestamp
+import com.google.gson.JsonArray
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import java.io.FileNotFoundException
+import java.text.SimpleDateFormat
+import java.util.*
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionException
+
+
+private const val TITLE_FIELD_NAME = "full_title"
+private const val AUTHORS_FIELD_NAME = "authors"
+private const val FORMAT_FIELD_NAME = "physical_format"
+private const val ISBN13_FIELD_NAME = "isbn_13"
+private const val PUBLISHER_FIELD_NAME = "publishers"
+private const val PUBLISH_DATE_FIELD_NAME = "publish_date"
+private const val AUTHOR_NAME_FIELD_NAME = "name"
+
+private const val DATE_FORMAT = "MMM dd, yyyy"
+private const val ISBN13_FORMAT = """[0-9]{13}"""
+private const val ISBN10_FORMAT = """[0-9]{9}[0-9X]"""
+private const val ISBN_FORMAT = """($ISBN10_FORMAT)|($ISBN13_FORMAT)"""
+
+private const val OL_BASE_ADDR = """https://openlibrary.org"""
+
+/**
+ * An implementation of a book database based on the Open Library online database
+ * */
+class OLBookDatabase(private val url2json : (String) -> CompletableFuture<JsonElement>) : BookDatabase {
+
+    override fun queryBooks(): BookQuery = OLBookQuery()
+
+    private inner class OLBookQuery : BookQuery {
```

Comment:
I am using a BookQuery for the `BooksFilterteringActivity` so I think we should keep this class public? Or do you know if I can create instance of OLBookQuery otherwise?

Reply:
you can get an instance of a OLBookQuery via a OLBookDatabase instance